### DISCUSSION TITLE
IDE에서 타입스크립트 타입체크는 실시간으로 이루어지는데, 그렇다면 컴파일과정이 계속 일어나는 건가?

### DISCUSSION THREAD

**Discussion by pipisebastian** [publdaze/js-study#38] [궁금증]
IDE에서 타입스크립트 타입체크는 실시간으로 이루어지는데, 그렇다면 컴파일 과정이 계속 일어나는 건가?

**Comment by publdaze** [comment]
- IDE에서 타입스크립트의 타입체크는 타입스크립트 컴파일러와 별개의 IDE 언어서비스로 제공되므로, 컴파일 과정이 계속 일어난다고 보기는 어려움
- IDE는 타입스크립트 언어서비스를 사용하여 실시간으로 타입을 검사하고, 이는 전체 컴파일 과정 없이 부분적인 코드 분석을 통해 이루어짐
- ([참고](https://code.visualstudio.com/docs/typescript/typescript-tutorial))
